### DISCUSSION TITLE
NodeJS: session set in supabase library on server-side environment

### DISCUSSION THREAD

**Discussion by Topener** [supabase/supabase#16646] [Questions] [answered]
Hi all!

I recently had another issue that was related, but I narrowed down the problem significantly I figured I better create a new question here.

Previous question: https://github.com/orgs/supabase/discussions/16571

I run NodeJS as middleware between Supabase and the client. This way the client doesn't have to work with supabase directly, and if we ever decide to switch authentication/database we don't have to alter the client.

We do use Supabase authentication email / JWT

This means, I refresh the tokens server-side. This goes well. The refresh token is retrieved, the new token+refresh token is generated, and I return this information back to the client.

However, whenever this code is called 
```js
const { data, error } = await supabase.auth.refreshSession({ refresh_token });
```

A session is stored serverside, allowing me to call this (I don't want this)

```js
console.log(await supabase.auth.getUser());
```

I figured this out because I have RLS enabled without policies and I use the `service_key`, but yet API requests are denied as they do not meet RLS after a `refreshSession` has been called. 

And I even have this configured:

```js
const supabase = createClient(process.env.SUPABASE_HOST, process.env.SUPABASE_KEY, {
    auth: {
        autoRefreshToken: false,
        persistSession: false,
        detectSessionInUrl: false,
    },
});
```

How do I prevent `supabase-js` from setting the session when I refresh the session, so I can keep calling the  database properly?

My workaround, which I know isn't a good fix, is calling`signOut` like this:
```js
    const { data, error } = await supabase.auth.refreshSession({ refresh_token });
    await supabase.auth.signOut();
```

During development, this will work just fine for now... but how do I fix this permanently?

**Comment by GaryAustin1** [comment, marked as answer]
Use two clients.  One to do service_role and another to refresh sessions.

**Reply by Topener** [reply]
Omg of course. I’ve been overthinking this waaay to much
